python/ray/tune/trial_runner.py

```diff
@@ -56,14 +56,17 @@ def step(self):
 
         if self._can_launch_more():
             self._launch_trial()
-        elif self._pending:
+        elif self._running:
             self._process_events()
         else:
             for trial in self._trials:
                 if trial.status == Trial.PENDING:
                     assert self._has_resources(trial.resources), \
                         ("Insufficient cluster resources to launch trial",
                          (trial.resources, self._avail_resources))
+                elif trial.status == Trial.PAUSED:
```

Comment:
I think you also need to add a check for the PAUSED state in `is_finished()`.

Reply:
Done.